test/soake2erekt/kafkasource_soak_test.go

```diff
@@ -0,0 +1,334 @@
+						return
+					}
+
+					uResources, found, err := unstructured.NestedSlice(u, "resources")
+					if !found {
+						// could be still empty?
+						continue
+					}
+					if err != nil {
+						t.Errorf("error getting .resources from %s configmap: %v", cm.Name, err)
+					}
+
+					for _, uResource := range uResources {
+						uResourceMap, ok := uResource.(map[string]interface{})
+						if !ok {
+							t.Errorf("unexpected type of a `resources` item in %s configmap", cm.Name)
+							continue
+						}
+						uNamespace, found, err := unstructured.NestedString(uResourceMap, "reference", "namespace")
+						if !found {
+							// could be not set?
+							continue
+						}
+						if err != nil {
+							t.Errorf("error getting .reference.namespace from %s configmap: %v", cm.Name, err)
+						}
+
+						// t.Logf("XXX reference namespace: %s", uNamespace)
+
+						if uNamespace == ns {
+							t.Errorf("Found reference to a resource in the test namespace %q in the %s configmap", uNamespace, cm.Name)
+						}
+					}
+				}
+			}
+		}
+	})
+
+	return f
+}
+
+/*
+TestKafkaSourceStableSoak on each iteration,
+sends an event to a kafkasource
+verifies an event is received
+*/
+func TestKafkaSourceStableSoak(t *testing.T) {
+	t.Parallel()
+
+	topicPrefix := feature.MakeRandomK8sName("topic")
+	kafkaSinkPrefix := topicPrefix
+	kafkaSourceName := feature.MakeRandomK8sName("kafkaSource")
+	senderPrefix := feature.MakeRandomK8sName("sender")
+
+	namesFn := func(env SoakEnv) kafkaSourceScenarioNames {
+		return kafkaSourceScenarioNames{
+			receiver:    "receiver",
+			sender:      fmt.Sprintf("%s-%d-%d", senderPrefix, env.CopyID(), env.Iteration()),
+			kafkaTopic:  fmt.Sprintf("%s-%d", topicPrefix, env.CopyID()),
+			kafkaSink:   fmt.Sprintf("%s-%d", kafkaSinkPrefix, env.CopyID()),
+			kafkaSource: kafkaSourceName,
+		}
+	}
+
+	soakTest := SoakTest{
+		NamespacePrefix: "test-kafka-source-stable-",
+		SetupFn: func(ctx context.Context, env environment.Environment, t *testing.T) {
+			// TODO: These are actually equivalent to just env.Test(ctx, t, eventshubReceiverFeature(namesFn(SoakEnvFromContext(ctx)))) , so not sure if we should bother...
+			RunSoakFeatureFnWithMapping(ctx, env, t, eventshubReceiverFeature, namesFn)
+			RunSoakFeatureFnWithMapping(ctx, env, t, kafkaSourceScenarioTopicAndSinkSetupFeature, namesFn)
+			RunSoakFeatureFnWithMapping(ctx, env, t, kafkaSourceScenarioInstallKafkaSourceFeature, namesFn)
+			RunSoakFeatureFnWithMapping(ctx, env, t, kafkaSourceScenarioIsReadyKafkaSourceFeature, namesFn)
+		},
+		IterationFn: func(ctx context.Context, env environment.Environment, t *testing.T) {
+			RunSoakFeatureFnWithMapping(ctx, env, t, kafkaSinkSendFeature, namesFn)
+			RunSoakFeatureFnWithMapping(ctx, env, t, verifySingleEventReceivedFeature, namesFn)
+
+			// we just want to verify the source can send/receive events throughout the soak test, so let it rest here for a while
+			time.Sleep(1 * time.Second)
```

Comment:
Not sure I understand the meaning of this Sleep. We're running in iterations so each iteration should verify receiving its own event.

Reply:
I mean, the `.AtLeast(atLeast)` function which is used to verify the received event already waits for some time for the event to be present.

Reply:
we don't want this iteration cycle to be "as fast as it can", the sleep is there for the thing to not be a stress test.

Reply:
Does the 1 second make any functional difference? It seems too short to make any effect. IMO, this is not a stress test because we're always waiting for the event to be received, so the system has enough time to send/receive event before next iteration.